lib/spack/docs/contexts.rst

```diff
@@ -0,0 +1,63 @@
+.. _contexts:
+
+Contexts
+============
+
+A context is used to group together a set of specs. This description
+can be used to replicate a set of installed packages on other systems.
+It can also be used to collect the environment modifications necessary
+to expose that subset of packages to a user.
+
+You start by creating the context:
+
+.. code-block:: console
+
+   $ spack context create c1
+
+Then you add some specs to it and concretize/install them:
+
+.. code-block:: console
+
+   $ spack context add c1 mpileaks
+   $ spack context add c1 python
+   $ spack context concretize c1
+   $ spack context install c1
+
+The context is stored in a file that can be relocated to another
+system. You can install all the packages in the context with the
```

Comment:
from 

> This does not install any concretized specs, but it does update the
 +saved context object.

do I understand this right that one does not have to re-concretize the context when relocating? 

What happens if version of compiler is different on another system? Or it's `ubuntu` instead of `centos`? Presumably relocation is only possible to the exactly equivalent system. If so, maybe mention it briefly?

Reply:
> do I understand this right that one does not have to re-concretize the context when relocating?

yes

> What happens if version of compiler is different on another system? Or it's ubuntu instead of centos?

As of now there would not be a failure until the user attempted to do `spack context install`. That being said there is (a currently undocumented) `spack context relocate` command which nullifies the compiler/os (leaving all other attributes as-is) and re-concretizes.

> Presumably relocation is only possible to the exactly equivalent system. If so, maybe mention it briefly?

Agreed (and I should mention `spack relocate` as well)